freeplane_plugin_script/src/main/java/org/freeplane/plugin/script/proxy/BorderProxy.java

```diff
@@ -0,0 +1,46 @@
+package org.freeplane.plugin.script.proxy;
+
+import org.freeplane.core.util.ColorUtils;
+import org.freeplane.features.map.NodeModel;
+import org.freeplane.features.nodestyle.NodeBorderModel;
+import org.freeplane.features.nodestyle.NodeStyleController;
+import org.freeplane.features.nodestyle.mindmapmode.MNodeStyleController;
+import org.freeplane.features.styles.LogicalStyleController;
+import org.freeplane.plugin.script.ScriptContext;
+
+import java.awt.Color;
+
+public class BorderProxy extends AbstractProxy<NodeModel> implements Proxy.Border {
+    BorderProxy(NodeModel delegate, ScriptContext scriptContext) {
+        super(delegate, scriptContext);
+    }
+
+    private MNodeStyleController getStyleController() {
+        return (MNodeStyleController) NodeStyleController.getController();
+    }
+
+    @Override
+    public void setColor(Color color) {
+        getStyleController().setBorderColor(getDelegate(), color);
+    }
+
+    @Override
+    public void setColorCode(String rgbString) {
+        setColor(ColorUtils.stringToColor(rgbString));
+    }
+
+    @Override
+    public Color getColor() {
+        return getStyleController().getBorderColor(getDelegate(), LogicalStyleController.StyleOption.FOR_UNSELECTED_NODE);
+    }
+
+    @Override
+    public String getColorCode() {
+        return ColorUtils.colorToString(getColor());
+    }
+
+    @Override
+    public boolean isColorSet() {
+        return NodeBorderModel.getModel(getDelegate())==null?false:(NodeBorderModel.getModel(getDelegate()).getBorderColor() != null);
```

Comment:
The delegate can not be null. Why do you check it?

Reply:
I check if there is a Model assigned to that delegate.

Like this:
https://github.com/freeplane/freeplane/blob/6bf842791d968c9e2a35c4432627d37e2a1ac9d2/freeplane/src/main/java/org/freeplane/features/nodestyle/NodeBorderModel.java#L62-L65

As far as I understood, if the node has no border style assigned, it won't have a NodeBorderModel created:

Example:
Node without NodeBorderModel created:
![image](https://github.com/freeplane/freeplane/assets/35700575/85c6052e-2398-440d-9c5f-15ee78aac654)

Node border format was modified for this node --> FP creates a NodeBorderModel for it:
![image](https://github.com/freeplane/freeplane/assets/35700575/29844dd0-7dd5-4872-8cc1-992fe5c934c7)

Reply:
getDelegate() which you compare to null is a NodeModel, isn't it?

Reply:
yes, it is.

I compare the associated NodeBorderModel to null , NOT the delegate (NodeModel)

the code:
```java
return NodeBorderModel.getModel(getDelegate())==null ? false : (Nod....
```

can also be written as:
```java
NodeModel myNode = getDelegate();
NodeBorderModel nodeBorderModel = NodeBorderModel.getModel(myNode);
return nodeBorderModel==null ? false : (Nod....
```

I don't know if I understand your question correctly. I hope I had answered it.

Best regards,

edo

Reply:
Yes, then I misunderstood the code.
Thank you for the clarification.
By the way, instead of using 
````
NodeBorderModel.getModel(getDelegate())==null?false:(NodeBorderModel.getModel(getDelegate()).getBorderColor() != null)
````

I would probably write

````
       NodeBorderModel border = NodeBorderModel.getModel(getDelegate());
       return border!=null && border.getBorderColor() != null;
````

to avoid duplication and to simplify the expression.